### PR TITLE
Mincost jump version

### PR DESCRIPTION
**Pull request by matbesancon**
For now we'll use JuMP, one pain is that the numeric type is forced to be Float64. Not a big deal as long as there are no type-generic LP solver

### CODE REVIEW

**Comment by ericphanson** [comment]
> there are no type-generic LP solver

Tulip.jl?

File: test/mincost.jl [hunk 1]
```diff
@@ -14,83 +16,67 @@ using SparseArrays: spzeros
...
-    @test flow[5,1] == 1
-    @test flow[5,2] == 1
-    @test flow[3,6] == 1
-    @test flow[4,6] == 1
-    @test flow[1,4] == 1
-    @test flow[2,4] == 0
-    @test flow[2,3] == 1
-    @test flow[1,3] == 0
-    @test sum(diag(flow)) == 0
+    o = with_optimizer(Clp.Optimizer)
```
Comment (ericphanson) [inline test/mincost.jl:30]: This was deprecated in JuMP 0.21 I think; but in JuMP 0.21 I believe you can just pass `Clp.Optimizer` itself.

**Comment by matbesancon** [comment]
You're right on Tulip I think, but not all real types can be used because it relies on non-generic LinearAlgebra if I'm not mistaken, @mtanneau is it correct?

**Comment by ericphanson** [comment]
I believe that used to be true, but he added support for more generic linear algebra via LDLFactorizations, and now at least BigFloat's can be used (as shown in the readme).

**Comment by matbesancon** [comment]
I think at some point this will be fixed at the JuMP level

**Comment by ericphanson** [comment]
That would be great! To be clear, I wasn't really suggesting rewriting it with MOI, just pointing out that there is a generic solver :). I made the other choice in Convex.jl, but that's since I really wanted to solve some SDPs in high-precision, and I think either option is pretty defensible.

**Comment by matbesancon** [comment]
no trouble, and yes I'm impatiently waiting to get eltype-generic linear and MIP solvers for many aspects

**Comment by matbesancon** [comment]
closes #16

**Comment by mtanneau** [comment]
> not all real types can be used because it relies on non-generic LinearAlgebra

Any real type should work. If it's not Float64, then the linear algebra is generic and it will use LDLFactorizations.jl (if your matrix is sparse) or generic Cholesky (if your matrix is dense).

That being said, for min-cost flow, network simplex is likely to be much more efficient that interior-point.

**Comment by matbesancon** [comment]
yes, still good to know, say you want to solve a symbolic flow, or differentiate through a network flow, etc

**Pull request merged**